integration/common/openlineage/common/provider/dbt.py

```diff
@@ -373,139 +380,135 @@ def parse_test(
                 assertions=assertions[name]
             )
 
-            namespace, name, _ = self.extract_dataset_data(ModelNode(node), has_facets=False)
+            namespace, name, _ = self.extract_dataset_data(
+                ModelNode(node),
+                assertion_facet,
+                has_facets=False
+            )
 
             job_name = f"{node['database']}." \
                 f"{node['schema']}." \
                 f"{self.removeprefix(node['unique_id'], 'test.')}"
 
             run_id = str(uuid.uuid4())
-            starts.append(RunEvent(
-                eventType=RunState.START,
-                eventTime=started_at,
-                run=Run(
-                    runId=run_id
-                ),
-                job=Job(
-                    namespace=self.job_namespace,
-                    name=job_name
-                ),
-                producer=self.producer,
-                inputs=[Dataset(namespace, name)],
-                outputs=[]
-            ))
-            completes.append(RunEvent(
-                eventType=RunState.COMPLETE,
-                eventTime=completed_at,
-                run=Run(
-                    runId=run_id,
-                    facets={
-                        "parent": self._dbt_run_metadata.to_openlineage(),
-                        "dbt_version": self.dbt_version_facet()
-                    }
-                ),
-                job=Job(
-                    namespace=self.job_namespace,
-                    name=job_name
-                ),
-                producer=self.producer,
-                inputs=[Dataset(namespace, name, facets={
+            events.add(self.to_openlineage_events(
+                "success",
+                started_at,
+                completed_at,
+                self.get_run(run_id),
+                Job(self.job_namespace, job_name),
+                [Dataset(namespace, name, facets={
                     "dataQualityAssertions": assertion_facet
                 })],
```

Comment:
We need to nail down what it means for a facet to be in `Dataset.facets` vs in `InputDataset.inputFacets` or `OutputDataset.outputFacets`. Currently the GreatExpectations integration attaches the `GreatExpectationsAssertionsDatasetFacet` and `DataQualityMetricsInputDatasetFacet` as input facets (see [here](https://github.com/OpenLineage/OpenLineage/blob/main/integration/common/openlineage/common/provider/great_expectations/action.py#L222-L227)), whereas the dbt integration attaches them as `Dataset.facets`. I see in the test case, the output dataset can have the assertions facet attached. I understand that's due to this refactoring, but it may be confusing for server-side consumers who don't know where to look for a given facet. Is the assertions facet really part of the output? or just describing the Dataset that was output? or is it describing the Dataset after the output was complete?

Reply:
>Currently the GreatExpectations integration attaches the GreatExpectationsAssertionsDatasetFacet and DataQualityMetricsInputDatasetFacet as input facets (see here), whereas the dbt integration attaches them as Dataset.facets.

It did attach them as input facets before, and still does when `dbt test` is a separate command.
Here, the creation of the dataset and testing it are parts of the same run - the dataset is in outputs of the job. 

If we really want to have it as an input facet, then the solution would be to split this run into two runs - the first one having this dataset as an output without the assertions, and having second run, with input dataset, with just the assertions. It would be like running `dbt run` and `dbt test` separately.

Reply:
We've discussed this offline with @collado-mike and decided to go with two separate events, each corresponding to `dbt run` and `dbt test` invocation.